src/views/abstractview.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <QWidget>
+
+/**
+ * @class AbstractViewModel class
+ *
+ * An abstract base class that defines 2 basic functions that
+ * views must define: resetView() and resetUI()
+ */
+class View: public QWidget
+{
+    Q_OBJECT
+    
+public:
+    explicit View(QWidget* parent = nullptr)
```

Comment:
A view should always have a parent, so putting a default parameter shouldn't happen.

Reply:
I just used the format that qt has so blame qt

Reply:
Fricken Qt 👎

Reply:
So no default then?

Reply:
No default
```cpp
QWidget* parent
```